internal/serverinstall/nomad.go

```diff
@@ -410,6 +452,18 @@ func (i *NomadInstaller) Uninstall(ctx context.Context, opts *InstallOpts) error
 	}
 
 	s.Update("Waypoint job and allocations purged")
+
+	vols, _, err := client.CSIVolumes().List(&api.QueryOptions{Prefix: "waypoint"})
+	for _, vol := range vols {
+		if vol.ID == "waypoint" {
+			s.Update("Destroying persistent CSI volume")
+			err = client.CSIVolumes().Deregister(vol.ID, false, &api.WriteOptions{})
+			if err != nil {
+				return err
+			}
+			s.Update("Successfully destroyed persistent volumes")
```

Comment:
Will there ever be more than 1 volume for Waypoint? maybe we could do a `break` here so we don't keep looping once we find the volume id we're after.